app/src/display/widgets/battery_status.c

```diff
@@ -33,16 +35,29 @@ void battery_status_init() {
     lv_style_set_text_line_space(&label_style, LV_STATE_DEFAULT, 1);
 }
 
+K_MUTEX_DEFINE(battery_status_mutex);
+
+struct {
+    uint8_t level;
+#if IS_ENABLED(CONFIG_USB)
+    bool usb_present;
+#endif
+} battery_status_state;
+
 void set_battery_symbol(lv_obj_t *label) {
     char text[2] = "  ";
-    uint8_t level = bt_bas_get_battery_level();
+    k_mutex_lock(&battery_status_mutex, K_FOREVER);
+
+    uint8_t level = battery_status_state.level;
 
 #if IS_ENABLED(CONFIG_USB)
-    if (zmk_usb_is_powered()) {
+    if (battery_status_state.usb_present) {
         strcpy(text, LV_SYMBOL_CHARGE);
```

Comment:
Not relevant to this change, and it's currently not an issue since the code can only write two characters to the string, but note that strcpy and strcat will happily overflow the buffer if one of those `LV_` constants happens to be longer than a single character or someone adds another strcat call later.